src/Geometry/conversions.jl

```diff
@@ -612,6 +619,12 @@ Curl is only defined for `CovariantVector`` field input types.
     ::Val{(1, 2)},
     ::Type{Covariant123Vector{FT}},
 ) where {FT} = Contravariant123Vector{FT}
+@inline curl_result_type(
+    ::Val{()},
+    ::Type{CovariantNullVector{FT}},
+) where {FT} = ContravariantNullVector{FT}
+@inline curl_result_type(::Val{()}, ::Type{<:CovariantVector{FT}}) where {FT} =
+    ContravariantNullVector{FT}
```

Comment:
You've covered null curls of null and non-null vectors, but you're missing a method for non-null curls of null vectors:
```
@inline curl_result_type(_, ::Type{CovariantNullVector{FT}}) where {FT} =
    ContravariantNullVector{FT}
```
You'll probably need to keep the method for null curls of null vectors to avoid introducing an ambiguity.